mne_bids/mne_bids.py

```diff
@@ -424,60 +493,66 @@ def raw_to_bids(subject_id, task, raw_file, output_path, session_id=None,
 
     make_dataset_description(output_path, name=" ",
                              verbose=verbose)
-    _sidecar_json(raw, task, manufacturer, data_meta_fname, kind, verbose)
+    _sidecar_json(raw, task, manufacturer, data_meta_fname, kind,
+                  eeg_reference, verbose)
     _channels_tsv(raw, channels_fname, verbose)
 
     events = _read_events(events_data, raw)
     if len(events) > 0:
         _events_tsv(events, raw, events_tsv_fname, event_id, verbose)
 
-    # for FIF, we need to re-save the file to fix the file pointer
-    # for files with multiple parts
-    if ext in ['.fif', '.gz']:
-        raw.save(raw_file_bids, overwrite=overwrite)
+    # Writing of neural data: Different for MEG and EEG
+    # Check if it is MEG (only writing to FIF)
+    # ----------------------------------------
+    if ext in allowed_extensions_meg:
+        # for FIF, we need to re-save the file to fix the file pointer
+        # for files with multiple parts
+        if ext in ['.fif', '.gz']:
+            raw.save(raw_file_bids, overwrite=overwrite)
+        else:
+            if os.path.exists(raw_file_bids):
+                if overwrite:
+                    os.remove(raw_file_bids)
+                    sh.copyfile(raw_file, raw_file_bids)
+                else:
+                    raise ValueError('"%s" already exists. Please set'
+                                     'overwrite to True.' % raw_file_bids)
+
+    # Check if it is EEG (preserving original file format)
+    # ----------------------------------------------------
+    elif ext in allowed_extensions_eeg:
+        if os.path.exists(raw_file_bids) and not overwrite:
```

Comment:
this part should be common to both EEG and MEG

Reply:
+1